GliaWidgets/Lib/Command/Command.swift

```diff
@@ -0,0 +1,67 @@
+/// Developer-friendly wrapper around a closure.
+/// It makes debugging easier by providing callee information.
+/// Note: since closure does not conform to `Equatable`, it will
+/// also be ignored in `Equatable` and `Hashable` implementations
+/// of `Command`.
+///
+/// Example of declaration and execution:
+/// ```
+/// let validateEmail = Command<String> { email in /* email validation goes here */ }
+/// validateEmail("email@example.test")
+/// ```
+struct Command<T>: Hashable {
+    let tag: String
+    let file: String
+    let function: String
+    let line: UInt
+    let closure: (T) -> Void
+
+    init(
+        tag: String = "",
+        file: StaticString = #file,
+        function: StaticString = #function,
+        line: UInt = #line,
+        closure: @escaping (T) -> Void
+    ) {
+        self.tag = tag
+        self.file = "\(file)"
+        self.function = "\(function)"
+        self.line = line
+        self.closure = closure
+    }
+
+    func execute(with value: T) {
+        closure(value)
+    }
+
+    func callAsFunction(_ value: T) {
+        execute(with: value)
+    }
+
+    static func == (lhs: Self, rhs: Self) -> Bool {
```

Comment:
Completely unrelated and of course nothing to do with your code, I have always despised this implementation by Apple. Who thought `lhs` and `rhs` were good names?

Reply:
I think it came from the languages like C/C++. LHS stands for left-hand side. I guess back in the day length of variables did matter. Also in math it is also commonly [used](https://en.wikipedia.org/wiki/Sides_of_an_equation).

Reply:
Ah maybe if you are an English speaker it makes more sense. I don't remember dealing with it in my limited experience with C++ so when I read it the first time I was dumbfounded.